client/post-editor/controller.js

```diff
@@ -107,11 +108,45 @@ function getPressThisContent( query ) {
 	return pieces.join( '\n\n' );
 }
 
+// TODO: REDUX - remove flux actions when whole post-editor is reduxified
+// Until the full migration, the Copy Post functionality needs to dispatch both Flux and Redux actions:
+// - (Flux) startEditingNew: to set the editor content;
+// - (Redux) editPost: to set every other attribute (in particular, to update the Category Selector, terms can only be set via Redux);
+// - (Flux) edit: to reliably show the updated post attributes before (auto)saving.
+function startEditingPostCopy( siteId, postToCopyId, context ) {
+	wpcom.site( siteId ).post( postToCopyId ).get().then( postToCopy => {
+		const postAttributes = pick(
+			postToCopy,
+			'canonical_image',
+			'content',
+			'excerpt',
+			'featured_image',
+			'format',
+			'metadata',
+			'post_thumbnail',
+			'terms',
+			'title',
+			'type'
+		);
+		postAttributes.tags = map( postToCopy.tags, 'name' );
+
+		actions.startEditingNew( siteId, {
+			type: 'post',
+			content: postToCopy.content,
+		} );
+		context.store.dispatch( editPost( siteId, null, postAttributes ) );
+		actions.edit( postAttributes );
+	} ).catch( () => {
+		actions.startEditingNew( siteId, { type: 'post' } );
```

Comment:
This is what happens now when there is trouble loading the post to copy.

![screenshot-2016-08-22-21 16 17](https://cloud.githubusercontent.com/assets/2070010/17868336/c8dd855c-68ae-11e6-9dd5-f138ec08cb7f.png)